pkg/acctest/test.go

```diff
@@ -277,3 +278,30 @@ func CompositeCheckDestroy(funcs ...func(state *terraform.State) error) func(sta
 		return nil
 	}
 }
+
+func DeleteUser(t *testing.T, name string) error {
+	restyClient := GetTestResty(t)
+	_, err := restyClient.R().Delete(user.UsersEndpointPath + name)
+
+	return err
+}
+
+func CreateUserUpdatable(t *testing.T, name string, email string) {
+	userObj := user.User{
+		Name:                     name,
+		Email:                    email,
```

Comment:
Since this is for testing, it's best to randomize the username in someway. So, the input parameter could say be the prefix, maybe indicating the test case that created it, and then this function and generate a randomized suffix, and then the function returns the username (or better yet, the whole user struct) generated so that it can be deleted later or interrogated

This is really the only way to test a statefull app, and we use it everywhere in the code base. The reasoning is simple: If a test fails to clean up after itself, due to error, network issue, etc, then subsequent tests will still run just fine because, due to randomization, they won't have a resource collision with previous runs. You could randomize it in your use case, but then you'd have to do it everytime. Best to do it here.